src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala

```diff
@@ -360,28 +365,27 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with Stash {
             if (currentBeamVehicle.isBEV | currentBeamVehicle.isPHEV) {
               stall.chargingPointType match {
                 case Some(_) =>
-                  handleStartCharging(tick, currentBeamVehicle)
-                  if (ChargingPointType.getChargingPointInstalledPowerInKw(
-                        currentBeamVehicle.stall.get.chargingPointType.get
-                      ) > 20.0) {
-                    val (sessionDuration, energyDelivered) =
-                      currentBeamVehicle.refuelingSessionDurationAndEnergyInJoules()
-                    log.debug(
-                      "scheduling EndRefuelSessionTrigger at {} with {} J to be delivered, triggerId: {}",
-                      tick + sessionDuration.toInt,
-                      energyDelivered,
-                      triggerId
-                    )
-                    scheduler ! ScheduleTrigger(
-                      EndRefuelSessionTrigger(
-                        tick + sessionDuration.toInt,
-                        tick,
-                        energyDelivered,
-                        currentBeamVehicle
-                      ),
-                      self
-                    )
+                  log.debug("Sending ChargingPlugRequest to chargingNetworkManager at {}", tick)
+                  try {
+                    Await
```

Comment:
Why `Await`? That will ultimately bottleneck the performance - you could just as easily handle the logging debug/error messages from an `onComplete` or the like

Reply:
it is not going to be a bottleneck. It's just a way to synchronize the the workflow and avoid juggling between states.